projects/plugins/jetpack/sal/class.json-api-post-base.php

```diff
@@ -422,7 +422,7 @@ protected function get_post_link( $path = null ) {
 	public function get_publicize_urls() {
 		$publicize_urls = array();
 		$publicize      = get_post_meta( $this->post->ID, 'publicize_results', true );
-		if ( $publicize ) {
+		if ( is_array( $publicize ) ) {
```

Comment:
The `get_post_meta` call has been passed `true` as the third argument, so if the value of the meta field is true this will return the meta field as a string. Any posts generating this result - a string - would have led to an empty array being returned as we never enter the foreach loop.
However if we manipulate that string data, we can unserialize and process in the same way as the arrays (with the same format).
@Automattic/triforce (for anyone more familiar with Publicize) - do you know if there is any reason why we aren't processing any string results from `get_post_meta`, for `get_publicize_urls`? It's 'always' been this way from what I can see, so it would seem it isn't a noticeable issue.

Reply:
This is probably a tech debt that should be fixed. I am not sure if that `publicize_results` meta is even set or used anywhere. We now use `_publicize_shares` meta to store the shared posts information.

Reply:
I think it's OK to make any changes/improvements to this part of the code/logic.

Reply:
Excellent, I've added a change now to include unserialization.